sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -1628,12 +1628,12 @@ ssize_t gcoap_req_send_tl(const uint8_t *buf, size_t len,
 
 int gcoap_resp_init(coap_pkt_t *pdu, uint8_t *buf, size_t len, unsigned code)
 {
-    if (coap_get_type(pdu) == COAP_TYPE_CON) {
-        coap_hdr_set_type(pdu->hdr, COAP_TYPE_ACK);
-    }
-    coap_hdr_set_code(pdu->hdr, code);
+    int header_len = coap_build_reply(pdu, code, buf, len, 0);
 
-    unsigned header_len  = coap_get_total_hdr_len(pdu);
+    /* request contained no-response option or error occurred */
```

Comment:
Nit: Commend is vague, the only error / negative return possible is `-ENOSPC`. But :woman_shrugging: